activestorage/app/models/active_storage/blob.rb

```diff
@@ -142,13 +146,14 @@ def service_headers_for_direct_upload
   #
   # Prior to uploading, we compute the checksum, which is sent to the service for transit integrity validation. If the
   # checksum does not match what the service receives, an exception will be raised. We also measure the size of the +io+
-  # and store that in +byte_size+ on the blob record.
+  # and store that in +byte_size+ on the blob record. The content type is automatically extracted from the +io+ unless
+  # the user has specified a +content_type+ and set the +extract_content_type_from_io+ as false.
   #
   # Normally, you do not have to call this method directly at all. Use the factory class methods of +build_after_upload+
   # and +create_after_upload!+.
-  def upload(io)
+  def upload(io, extract_content_type_from_io = true)
```

Comment:
Let’s use a keyword argument here. Name it `identify` to match `build_after_upload` and `create_after_upload`.

Reply:
Making it a keyword argument makes it much clearer. Amended here e77f3db21842d20a489853b41eed4233c5394666